internal/header/accept_encoding.go

```diff
@@ -0,0 +1,75 @@
+/* Copyright 2019 Freerware
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package header
+
+import (
+	"sort"
+	"strings"
+)
+
+// AcceptEncoding represents the Accept-Encoding header.
+//
+// The "Accept-Encoding" header field can be used by user agents to
+// indicate what response content-codings (Section 3.1.2.1) are
+// acceptable in the response.
+type AcceptEncoding struct {
```

Comment:
can consider having `type AcceptEncoding []ContentCoding` instead, as the struct simply has this single property.

Reply:
done!